### PR TITLE
Feature/show subcategories on home page

### PR DESCRIPTION
**Pull request by BogdanLinnik**

### CODE REVIEW

File: app/views/pages/home/_categories.html.erb [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
-        <%= link_to category.name, {filter: { category: category.id }}, {class: "btn btn-outline-primary btn-lg col"} %>
-          <% unless category.children.empty? %>
-            <ul class='categories__subcategory'>
-            <% category.children.each do |subcategory| %>
-              <li><%= link_to subcategory.name, filter: { category: subcategory.id } %></li>
-            <% end %>
-            </ul>
+    <% @parent_categories.each do |category| %>
+      <ul class='categories__subcategory col mb-2'>
+        <% if @filtered_by&.==category.id %>
```
Comment (illia108) [inline app/views/pages/home/_categories.html.erb]: why `&` ?

File: app/controllers/pages_controller.rb [hunk 2]
```diff
@@ -28,6 +28,11 @@ def search
 
   def set_recipes
     @recipes = ArrangeRecipes.call(sort_params, filter_params[:filter]);
+    if filter_params[:filter]
```
Comment (illia108) [inline app/controllers/pages_controller.rb]: use `dig` instead of two ifs

File: app/views/pages/home/_categories.html.erb [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+          <%= link_to category.name, {filter: { category: category.id }}, class: "list-group-item active" %>
+        <% else %>
+          <%= link_to category.name, {filter: { category: category.id }}, {class: "list-group-item", style: "background-color: lightblue" }%>
+        <% end %>
+        <% unless category.children.empty? %>
+          <% category.children.each do |subcategory| %>
+        <% if @filtered_by&.==subcategory.id %>
```
Comment (illia108) [inline app/views/pages/home/_categories.html.erb]: same here

File: app/views/pages/home/_categories.html.erb (hunk 1 above)
Comment (BogdanLinnik) [reply app/views/pages/home/_categories.html.erb]: removed

**Pull request merged**